### ISSUE TITLE
What exactly is the model_final.pth? The best one or simply the last one?

### ISSUE THREAD

**Issue by gn1024** [facebookresearch/detectron2#1381]
First let me thank you for the excellent software.
I'm using it to train a custom COCO dataset for instance segmentation.
My training class is as follows (I added some evaluation).

```
class BirdsTrainer(DefaultTrainer):
    @classmethod
    def build_evaluator(cls, cfg, dataset_name, output_folder=None):
        if output_folder is None:
            output_folder = os.path.join(cfg.OUTPUT_DIR, 'evaluation')
        return COCOEvaluator(dataset_name, cfg, True, output_folder)

    @classmethod
    def test_with_TTA(cls, cfg, model):
        logger = logging.getLogger('detectron2.trainer')
        # In the end of training, run an evaluation with TTA
        # Only support some R-CNN models.
        logger.info('Running inference with test-time augmentation ...')
        model = GeneralizedRCNNWithTTA(cfg, model)
        evaluators = [
            cls.build_evaluator(cfg, name, output_folder=os.path.join(cfg.OUTPUT_DIR, 'inference_TTA'))
            for name in cfg.DATASETS.TEST
        ]
        res = cls.test(cfg, model, evaluators)
        res = OrderedDict({k + '_TTA': v for k, v in res.items()})
        return res
```

It works very well and generates a set of models ending with **model_final.pth**.
I am wondering if the **model_final.pth** is simply the last step model or the best one in terms of **loss** and **mAP**?
Thank you.

**Comment by ppwwyyxx** [comment]
"final" means the last.
